### PR TITLE
AEA-380 Tests for the code in the docs

### LINKED ISSUE #770: Tests for code in each doc.
**Is your feature request related to a problem? Please describe.**
Currently, code in docs is not tested.

**Describe the solution you'd like**
Add tests for all code in docs.

### PR DESCRIPTION
**Pull request by Totoual**
## Proposed changes

- [x] Multiplexer-standalone.md

## Fixes

#770 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_docs/test_standalone_transaction/test_standalone_transaction.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        """Test the last code block, that is the full listing of the demo from the Markdown."""
+        assert (
+            self.code_blocks[-1] == self.python_file
+        ), "Files must be exactly the same."
+
+    def test_run_agent(self, pytestconfig):
+        """Run the agent from the file."""
+        if pytestconfig.getoption("ci"):
+            pytest.skip("Skipping the test since it doesn't work in CI.")
+        run()
```
Comment (DavidMinarsch) [inline tests/test_docs/test_standalone_transaction/test_standalone_transaction.py]: you aren't asserting anything here. Why not capture the log output "The transaction digest is .." (see e.g. https://stackoverflow.com/questions/7472863/pydev-unittesting-how-to-capture-text-logged-to-a-logging-logger-in-captured-o)

File: docs/decision-maker-transaction.md [hunk 2]
```diff
@@ -1,170 +1,170 @@
 This guide can be considered as a part 2 of the <a href="/standalone-transaction/">the stand-alone transaction demo </a> we did in a previous guide. After the completion of the transaction,
 we get the transaction digest. With this we can search for the transaction on the <a href='https://explore-testnet.fetch.ai'>block explorer</a>. The main difference is that now we are going to use the decision-maker to settle the transaction.
 
-## Create the private keys
+Firstly import the libraries and the set the constant values.
```
Comment (DavidMinarsch) [inline docs/decision-maker-transaction.md]: First, import the libraries and set the constant values.

File: docs/decision-maker-transaction.md [hunk 3]
```diff
@@ -1,170 +1,170 @@
...
-
+``` python
+    # Set the AEA running in a different thread
+    try:
+        logger.info("STARTING AEA NOW!")
+        t = Thread(target=my_aea.start)
+        t.start()
+
+        # Let it run long enough to interact with the weather station
+        time.sleep(5)
```
Comment (DavidMinarsch) [inline docs/decision-maker-transaction.md]: I think 5 seconds is a bit short? Maybe give it more

File: docs/decision-maker-transaction.md [hunk 4]
```diff
@@ -182,20 +182,16 @@ class TransactionHandler(Handler):
         :return: None
         """
         tx_msg_response = cast(TransactionMessage, message)
-        self.context.logger.info(tx_msg_response)
+        logger.info(tx_msg_response)
```
Comment (DavidMinarsch) [inline docs/decision-maker-transaction.md]: @Totoual please revert to the `self.context.` approach which was introduced by Marco in a recent PR

File: docs/decision-maker-transaction.md [hunk 5]
```diff
@@ -182,20 +182,16 @@ class TransactionHandler(Handler):
...
         if (
             tx_msg_response is not None
             and tx_msg_response.performative
             == TransactionMessage.Performative.SUCCESSFUL_SETTLEMENT
         ):
-            self.context.logger.info(
-                "Transaction was successful."
-            )
-            self.context.logger.info(tx_msg_response.tx_digest)
+            logger.info("Transaction was successful.")
```
Comment (DavidMinarsch) [inline docs/decision-maker-transaction.md]: see above

File: docs/decision-maker-transaction.md [hunk 6]
```diff
@@ -329,20 +333,16 @@ class TransactionHandler(Handler):
...
```
Comment (DavidMinarsch) [inline docs/decision-maker-transaction.md]: see above

File: docs/decision-maker-transaction.md (hunk 4 above)
Comment (Totoual) [reply docs/decision-maker-transaction.md]: It results to an error that the `str has no info`

Comment (Totoual) [reply docs/decision-maker-transaction.md]: Actually it is not possible to do this based on the way I am instantiating the TransactionHandler. 
`tx_handler = TransactionHandler(skill_context="skill_context", name="fake_skill")`
The way around would be to create a skill_context and pass it to the handler.

Comment (DavidMinarsch) [reply docs/decision-maker-transaction.md]: ah ok, no fine, totally ok then

**Pull request merged**